src/runtime/DateObject.cpp

```diff
@@ -1419,4 +1419,50 @@ void* DateObject::operator new(size_t size)
     }
     return GC_MALLOC_EXPLICITLY_TYPED(size, descr);
 }
+
+Value DateObject::makeDay(ExecutionState& state, const Value& year, const Value& month, const Value& day)
+{
+    if (!std::isfinite(year.asDouble()) && !std::isfinite(month.asDouble()) && !std::isfinite(day.asDouble())) {
+        return Value(std::numeric_limits<double>::quiet_NaN());
+    }
+
+    double y = year.asDouble();
+    double m = month.asDouble() - 1;
+    double dt = day.asDouble();
+
+    time64_t result = 0;
+
+    double ym = y + std::floor(m / 12);
+
+    if (!std::isfinite(ym)) {
+        return Value(std::numeric_limits<double>::quiet_NaN());
+    }
+
+    result += const_Date_msPerDay * 365.2425 * (ym - 1970) + 2629743000 * ((int)m % 12);
```

Comment:
These constants are hard to understand: 365.2425 / 2629743000

Maybe some named constants could be introduced for them, and they could be computed, e.g.: `(24*60*60)` or something like that. Also large constants such as 2629743000 might need explicit type (e.g. uint64_t) to avoid miscomputation.